### PR TITLE
DEPR: Remove literal string input for read_html

### LINKED ISSUE #53767: [DEPR]: Remove literal string/bytes input from `read_excel`, `read_html`, and `read_xml`
Following on from #52271 and #53409 (and to close out #5924), here is the state of play for IO routines in terms of what they accept as input 

- `read_clipboard`: doesn't take input argument
- `read_csv`: pathlike, BinaryIO, StringIO
- `read_excel`: pathlike, BinaryIO, engine-specific objects, literal bytes
- `read_feather`: pathlike, BinaryIO
- `read_fwf`: pathlike, BinaryIO, StringIO
- `read_gbq`: SQL query (str)
- `read_hdf`: pathlike, pytables.HDFStore
- `read_html`: pathlike, BinaryIO, StringIO, literal bytes, literal string
- `read_json`: pathlike, BinaryIO, StringIO, literal string (deprecated in #53409)
- `read_orc`: pathlike, BinaryIO
- `read_parquet`: pathlike, BinaryIO
- `read_pickle`: pathlike, BinaryIO
- `read_sas`: pathlike, BinaryIO
- `read_spss`: pathlike (pyreadstat doesn't allow reading from a buffer)
- `read_sql_query`: SQL query (str)
- `read_sql_table`: name of database table (str)
- `read_stata`: pathlike, BinaryIO
- `read_table`: pathlike, BinaryIO, StringIO
- `read_xml`: pathlike, BinaryIO, StringIO, literal bytes, literal string

So some, but not all, of the textual formats support reading literally, some support reading bytes as well as strings, and some of the binary formats support reading literal bytes.

Having deprecated literal input to `read_json` I suggest it makes sense to therefore also do so for:

- `read_excel`
- `read_html`
- `read_xml`

### PR DESCRIPTION
**Pull request by rmhowe425**
- [X] closes #53767
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [X] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by rmhowe425** [comment]
@mroeschke PR is ready for inspection!

File: doc/source/whatsnew/v2.1.0.rst [hunk 1]
```diff
@@ -298,13 +298,15 @@ Deprecations
 - Deprecated constructing :class:`SparseArray` from scalar data, pass a sequence instead (:issue:`53039`)
 - Deprecated falling back to filling when ``value`` is not specified in :meth:`DataFrame.replace` and :meth:`Series.replace` with non-dict-like ``to_replace`` (:issue:`33302`)
 - Deprecated literal json input to :func:`read_json`. Wrap literal json string input in ``io.StringIO`` instead. (:issue:`53409`)
+- Deprecated literal string/bytes input to :func:`read_html`. Wrap literal string/bytes input in ``io.StringIO`` instead. (:issue:`53767`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:301]:
```suggestion
- Deprecated literal string/bytes input to :func:`read_html`. Wrap literal string/bytes input in ``io.StringIO``/``io.BytesIO`` instead. (:issue:`53767`)
```

File: pandas/io/html.py [hunk 2]
```diff
@@ -1178,6 +1183,15 @@ def read_html(
 
     io = stringify_path(io)
 
+    if isinstance(io, str) and "\n" in io:
```
Comment (mroeschke) [inline pandas/io/html.py:1186]: Is `\n` a reliable way detect if it's literal html?

File: pandas/io/html.py [hunk 3]
```diff
@@ -1178,6 +1185,15 @@ def read_html(
 
     io = stringify_path(io)
 
+    if isinstance(io, str) and not is_file_like(io) and "\n" in io:
```
Comment (mroeschke) [inline pandas/io/html.py:1188]: Could use the same type of checks like in the xml PR and remove the `\n` check?

**Comment by rmhowe425** [comment]
@mroeschke PR is ready for inspection.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rmhowe425

**Pull request merged**